Added VirtualCapacity to MachineClass NodeTemplate

elankath opened the pull request:
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:
<!--  Write your release note:
1. Enter your release note in the below block.
2. If no release note is required, just write "NONE" within the block.

Format of block header: <category> <target_group>
Possible values:
- category:       breaking|feature|bugfix|doc|other
- target_group:   user|operator|developer|dependency
-->
```feature operator
Machine Class now has VirtualCapacity that maps to WorkerPool's NodeTemplate VirtualCapacity.
```

ialidzhikov committed:
Remove the YEAR from the license boilerplate (#669)

Signed-off-by: ialidzhikov <i.alidjikov@gmail.com>

himanshu-kun committed:
Retry with Backoff for machineset finalizer update (#668)

Co-authored-by: Samarth Deyagond <samarth.deyagond@sap.com>

himanshu-kun committed:
Trigger orphan collection on OutOfRange machine status (#667)

gardener-robot-ci-1 committed:
Release v0.43.0

gardener-robot-ci-2 committed:
Prepare next Dev Cycle v0.44.0-dev

n-boshnakov committed:
Fixed broken links (#672)

ialidzhikov committed:
Log Node conditions when Machine is considered as unhealthy (#676)

Signed-off-by: ialidzhikov <i.alidjikov@gmail.com>

himanshu-kun committed:
nodeName as a colmn (#677)

himanshu-kun committed:
removed preserveUnknown field (#670)

Commit:
Improve documentation (#673)

* improve documentation

* added make rule to generate api docs

himanshu-kun committed:
CreateMachine logs improved (#678)

n-boshnakov committed:
Fixed broken links - 02.2022 (#682)

* Fixed broken links

* Fixed additional broken links

himanshu-kun committed:
typo corrected

unmarshall committed:
Merge pull request #685 from himanshu-kun/scale-down-disable-during-rollout

Scale down annotation value corrected

ialidzhikov committed:
Fix a typo (#690)

Signed-off-by: ialidzhikov <i.alidjikov@gmail.com>

himanshu-kun committed:
Limit #machines marked failed at a time due to being Unhealthy (#683)

Co-authored-by: Samarth Deyagond <sdeyagond@microsoft.com>
Co-authored-by: Martin Weindel <martin.weindel@sap.com>

Commit:
Release v0.44.0

gardener-robot-ci-1 committed:
Prepare next Dev Cycle v0.45.0-dev

himanshu-kun committed:
FAQ updated related to garbage collection (#695)


Co-authored-by: Ashwani Kumar <ash.kumar@sap.com>

Co-authored-by: Ashwani Kumar <ash.kumar@sap.com>

Mkmittal committed:
Adds permit testing testcases (#691)

Co-authored-by: Himanshu Sharma <79965161+himanshu-kun@users.noreply.github.com>

kon-angelo committed:
refactor annotations

ialidzhikov committed:
Fix typos

Signed-off-by: ialidzhikov <i.alidjikov@gmail.com>

himanshu-kun committed:
stale permit testcase time increased

Commit:
add/remove annotations logic added

kon-angelo committed:
update stale node check to check if node names are equal

Commit:
Automating local setup

Kostov6 committed:
Update manifest.yaml (#689)

Mkmittal committed:
Update local_setup.sh to take provider name as flag (#712)

himanshu-kun committed:
alpine and golang image updated (#713)

Mkmittal committed:
Update Kubernetes dependency versions to v1.21.12 (#719)

n-boshnakov committed:
Updated content structure (#710)

* Updated content structure

* Added personas

n-boshnakov committed:
Fixed broken links (#723)

Mkmittal committed:
Upgrade k8s dependencies to v1.22.9(the latest patch) (#721)

* Api Violation correction done

Mkmittal committed:
Added docs/todo/outline.md  (#700)

gardener-robot-ci-2 committed:
Release v0.45.0

Commit:
Prepare next Dev Cycle v0.46.0-dev

n-boshnakov committed:
Removed overrides from manifest file (#726)

Mkmittal committed:
Added more unit Tests for deployment controller(#725)

timuthy committed:
Build Multi-Arch Images 📦 (#732)

* Unifi local and Docker build step

* Let pipeline build multi-arch images

himanshu-kun committed:
Update revendor rule to be more descriptive

himanshu-kun committed:
Mark machine failed during rollout also (#733)

n-boshnakov committed:
Fixed broken links (#735)

himanshu-kun committed:
removed test case (#736)

dimityrmirchev committed:
Change base image from alpine to distroless (#734)

acumino committed:
migrate from endpointleases to leases (#711)

Co-authored-by: Himanshu Sharma <himanshu.sharma10@sap.com>

gardener-robot-ci-2 committed:
Release v0.46.0

gardener-robot-ci-1 committed:
Prepare next Dev Cycle v0.47.0-dev

rishabh-11 committed:
update clusterroles and clusterrolebindings to v1 (#738)

shafeeqes committed:
Handle `PodDisruptionBudget` `policy/v1` (#744)

Gerrit91 committed:
Support URL encoded bootstrap token replacement (#743)

Commit:
Release v0.47.0

Commit:
Prepare next Dev Cycle v0.48.0-dev

rishabh-11 committed:
Add nodeName and providerID column while listing machines(s) (#746)

* display providerID as a column name for machines with -owide
* display node name without -owide

rishabh-11 committed:
Added short names for MCM related resources (#747)

rishabh-11 committed:
Change short name `machcls` to `machclass` (#748)

rishabh-11 committed:
Update short names for MCM crds (#749)

rishabh-11 committed:
Updated go version to 1.19 (#753)

himanshu-kun committed:
Print errors properly (#759)

himanshu-kun committed:
proposal added for update-tags

himanshu-kun committed:
Update progressConditions even when no progressDeadlineSeconds configured(infinite deadline) (#762)

rishabh-11 committed:
Removes node field from machine status (#745)

himanshu-kun committed:
Upgrade to go1.19.4 (#766)

himanshu-kun committed:
proposal enhanced after discussion with @unmarshall

himanshu-kun committed:
Apply suggestions from code review

Co-authored-by: Madhav Bhargava <madhav.bhargava@sap.com>

himanshu-kun committed:
addressed review comments

himanshu-kun committed:
Scale-up only new machineSet while scale-down all active mSs proportionally (#765)

* scale-up only new mS while scale-down all active mSs proportionally

Co-authored-by: Madhav Bhargava <madhav.bhargava@sap.com>

gardener-robot-ci-2 committed:
Release v0.48.0

gardener-robot-ci-2 committed:
Prepare next Dev Cycle v0.49.0-dev

elankath committed:
Removes Deprecated In-tree Code(#769)

* Regeneration of pkg/client - clientset/informers/listenrs

* Added tags and docs/docs to gitignore

* Removal of dead/deprecated options for MCM binary

* Removal of deprecated api and its validation code

* Removal of provider specific CRD's, machine classes and openapi generated modification

* Removal of old machine controller code under `pkg/controller`

* Removal of driver migrate method, migration code. Marked safey options field as non-deprecated. Removed un-used code.

* Remove old machine-controller-manager-cli binary which is no longer used.

* Some more deprecated options and informers removed

* Unused constants and functions

* old drivers removed.

* Removed dead in-tree reference from readme's

* Removed in-tree refs, old secrets, old yaml and updated docs

* removed pv,pvc and secret lister/informers from MCM controller

* refactored controllers to use common worker in `util/worker/worker.go` and harmonized constants.

* removed old flags from deployment.yaml

* fixed import order and queue name in doc

* fixed import order in machinecontroller.go

* added docu comment for worker.DefaultMaxRetries

schrodit committed:
Fix machine creation bootstrap tokens expiration (#773)

himanshu-kun committed:
Updated code-owners, contributing file (#771)

elankath committed:
Remove secrets queue and secretSynced MCM usage (#780)

* remove secrets queue and secretSynced MCM usage

* fix test for secrets queue, secretSynced usage

SimonKienzler committed:
Consider critical-components-not-ready taint when turning machine to Running(#778)

* Consider critical-components-not-ready taint when checking machine health

* Implement PR feedback

* Add suggestions to comments

* Add test cases for taint scenarios

* Add extra test for unknown machine + tainted node

* refactored code

* vendored files

---------

Co-authored-by: Himanshu Sharma <himanshu.sharma10@sap.com>

dkistner committed:
Add cve categorization for mcm (#791)

The categorization was initially defined here and it transferred as defined to the mcm repository:
https://github.com/gardener/gardener-extension-provider-aws/pull/684
https://github.com/gardener/gardener-extension-provider-gcp/pull/546
https://github.com/gardener/gardener-extension-provider-azure/pull/621
https://github.com/gardener/gardener-extension-provider-openstack/pull/553
https://github.com/gardener/gardener-extension-provider-alicloud/pull/562
https://github.com/gardener/gardener-extension-provider-vsphere/pull/362

n-boshnakov committed:
Fixed broken links

elankath committed:
Merge pull request #797 from n-boshnakov/fix-broken-links-03.2023

Fixed broken links - 2023.03

ialidzhikov committed:
Fix the API docs generation (#800)

Co-authored-by: himanshu-kun <himanshu.sharma10@sap.com>

afritzler committed:
Fix linting issues in controller package (#794)

- Fix linting issues in controller package
- Fix unhandled errors
- Remove unused code

afritzler committed:
Bump `k8s.io/* ` and other dependencies (#792)

* Bump k8s.io/* dependencies

* add klog flags to flagset

* restructured import statements

* Add log flags initialization

* Revert "add klog flags to flagset"

This reverts commit 1c431ebda207767107bf32b70bd71b7b64a141cb.

---------

Co-authored-by: Himanshu Sharma <himanshu.sharma10@sap.com>

afritzler committed:
Fix linting issues in api package (#793)

afritzler committed:
Fix linting issues in `test`, `handlers` and `features` packages (#795)

* Fix linting issues in test, handlers and features package

* Remove dot imports

* Add not found matcher to framework cleanup

* Copy k8s matchers from gardener

unmarshall committed:
updated the proposal doc

unmarshall committed:
updated docs post discussion

himanshu-kun committed:
Edge cases corrected where all machineSets were scaling down to zero (#803)

* edge cases dealt with, unit tests added

* removed printf statements

* removed unused method

* minor fix

* error logging corrected

* added warning log

---------

Co-authored-by: Madhav Bhargava <madhav.bhargava@sap.com>

elankath committed:
updated toc

elankath committed:
added draft overview diagram

elankath committed:
correction draft overview diagram

unmarshall committed:
removed the diagram as it was not complete. description(text) is good enough for now

unmarshall committed:
Merge pull request #761 from himanshu-kun/update-tags-proposal

Proposal for hot-update of resources (Instance / NIC / Disk)

rishabh-11 committed:
Turn `CrashloopBackoff` machines to `Running` quicker (#806)

* fix reconcileMachineHealth for clbf machines

* address review comments

* make generate

* update docstring for clbf

* add comment

* add unit test for clbf to pending machine

himanshu-kun committed:
fix - add release note (#769)

himanshu-kun committed:
fix - add release note (#778)

gardener-robot-ci-2 committed:
Release v0.49.0

gardener-robot-ci-2 committed:
Prepare next Dev Cycle v0.50.0-dev

acumino committed:
Fix nil pointer panic (#814)

* Add missing parameter in log

* Drop unused function parameter

* Prevent nil pointer panic

* added log for missing mcs

* updated errorneous comments

---------

Co-authored-by: Himanshu Sharma <himanshu.sharma10@sap.com>

n-boshnakov committed:
Fix broken links - 2023.05 (#813)

* Fixed broken links

* Fixed numbering

* Removed broken image

- image has been removed in a previous PR, making the whole section unneeded

* Updated fixed links

rishabh-11 committed:
Keep `DesiredReplicas` annotation up-to-date even if no scaling happens (#821)

* keep desired replicas annotation up to date even if no scaling happens

* Update pkg/controller/deployment_sync_test.go

Co-authored-by: Himanshu Sharma <79965161+himanshu-kun@users.noreply.github.com>

---------

Co-authored-by: Himanshu Sharma <79965161+himanshu-kun@users.noreply.github.com>

himanshu-kun committed:
Refactor metrics code (#823)

* refactored var block and collector registration part

* refactored Collect() method for MC

* refactored Collect() method for MCM

* correction to helper function in MC metric update part

* made metric enumeration more intutive

* adds metric port for MC container

* removed cloud_api subsystem from MCM , kept in MC , removed promatu dependency, renamed functions,

* resolved lint errors

himanshu-kun committed:
Updated/added entries to FAQ (#819)

* updated/added entries to FAQ

* added note

* made corrections

jguipi committed:
Add metric to track Unhealthy Stale machine removal count (#808)

* refactored var block and collector registration part

* removed cloud_api subsystem from MCM , kept in MC , removed promatu dependency, renamed functions,

* stale machine counter metrics added

* made counter handling thread safe

* updated help message

* move variable in machineset.go

---------

Co-authored-by: Himanshu Sharma <himanshu.sharma10@sap.com>

afritzler committed:
Bump k8s.io/* deps to v0.27.2 (#820)

rishabh-11 committed:
upgrade golang version to v1.20.5 (#827)

n-boshnakov committed:
Updated manifest & added index files

n-boshnakov committed:
Fixed broken links

ashwani2k committed:
Merge pull request #830 from n-boshnakov/update-structure

Updated the docforge manifest & added index files

rishabh-11 committed:
include unavailable replicas in mcd status update check (#833)

vlerenc committed:
Introduce persona-based docs (#836)

ialidzhikov committed:
Nit: Fix the format verb (#831)

aaronfern committed:
Minor updates to integration tests doc (#843)

unmarshall committed:
Introduce new metrics to track API call duration and Update status.Status to capture underlying cause (#842)

* added histogram metrics to capture invocation duration of provide APIs

changed the name of the API request duration metric

added metric to capture overall duration for driver api calls

reverted back to using service label for provider api metric

added cause to status.Status

added WrapError function

corrected docstrings

corrected leftover docstring

Update pkg/util/provider/metrics/metrics.go

Co-authored-by: Himanshu Sharma <79965161+himanshu-kun@users.noreply.github.com>

Apply suggestions from code review

Co-authored-by: Himanshu Sharma <79965161+himanshu-kun@users.noreply.github.com>

addressed review comment - removed populating cause when in FromError when err is already status.Status

* added Driver failed API request counter

unmarshall committed:
Corrected local_setup.sh, added local_restore.sh, added .env file, added script to add license headers (#845)

rishabh-11 committed:
Introduced `errorCode` field in `LastOperation` struct (#851)

* add errorCode field in LastOperation struct

* address review comments

* address review comments part-2

elankath committed:
Fast drain + delete volume attachment for specific unhealthy nodes (#839)

* dont skip drain for unhealthy nodes

* Removed lastStateTransitionTime - undeeded for fix

* removed debug message

* corrected stage name

* reverted trivial changes to drain

* Addressed review comments and documentation

* fixed anchor link for new q in faq

* changed FAQ q to explain forceful deletion of unhealthy nodes

* removed excessive whitespace in faq

* del of vol attachments log machine name, minor faq update

* fixed unit tests after lastOperation.description adjust

unmarshall committed:
Updates to local setup scripts and introduce new local setup scripts for non-gardener use cases (#852)

* wip commit

* created separate script for non-gardener use case and also updated the doc

* added validation for kubeconfig expiry seconds

* addressed review comments

* minor corrections to the local_setup.md

* remove the info print statements from tools.mk

Commit:
Release v0.50.0

gardener-robot-ci-2 committed:
Prepare next Dev Cycle v0.51.0-dev

piyuagr committed:
Removed .env file  (#859)

piyuagr committed:
Updated Integration tests doc. (#861)

zkdev committed:
fix: Concourse build badge URL (#862)

afritzler committed:
Bump k8s.io/* deps to v0.28.2 (#858)

piyuagr committed:
Fix Rotation of log files in IT (#864)

* Rotation of log files is successfully happening on multiple runs of Integration Tests.

* Rotation of log files is successfully happening on multiple runs of Integration Tests.

* Fix Rotation of log Files in IT

* Fix for Rotation of Log File issue in IT.

* Fixed make check errors.

* Updated with Review comments and code changed

himanshu-kun committed:
Enhanced rate-limited removal logs (#867)

elankath committed:
Remove NotManagedByMCM when not applicable (#866)

* Remove NotManagedByMCM when not applicable

* enhanced unit test in case #AnnotateNodesUnmanagedByMCM

* log both node and machine name when removing NotManagedByMCM annotation

* review comment: leverage updateNodeWithAnnotations for removal

* fixed import reordering to original in cli options

piyuagr committed:
Remove Kubeconfig dependency for verification of control cluster in IT (#869)

* Removed the kubeconfig dependencies and updated the integration doc.

* Corrected semantics by Adding constant integration-test-cluster.

* Incorporated Review Comments.

* Corrected make check test.

* Moved from var to constant.

* Modified the comment for the new method.

piyuagr committed:
Updated to ginkgo v2 in mcm (#870)

* Updated to ginkgo v2 in mcm

* Added untracked files

* Moved ginkgo/ginkgo to ginkgo/v2.

* Incorporated Review Comments
.

himanshu-kun committed:
preliminary changes

himanshu-kun committed:
reconcile machineClass only on addition/deletion
of machines

himanshu-kun committed:
removed double check in secret reconciler

himanshu-kun committed:
small refactoring

himanshu-kun committed:
introduced machine conditions diff logging

himanshu-kun committed:
removed old in-tree related code

himanshu-kun committed:
adapted unit tests ot new const
errSuccessfulPhaseUpdate

piyuagr committed:
Removed old IT code from mcm. (#873)

* Removed old IT code from mcm.

* Removed the SKIP_INTEGRATION_TESTS variable declaration inside the test file.

rishabh-11 committed:
update go to 1.21.4 (#876)

himanshu-kun committed:
added new retry period ConflictRetry

himanshu-kun committed:
updated log level for better readability

himanshu-kun committed:
some more updated log levels

himanshu-kun committed:
adapted test cases

himanshu-kun committed:
no machine reconcile on status update

himanshu-kun committed:
updated log messages and used ConflictRetry

himanshu-kun committed:
make generate && make check

ccwienk committed:
Switch GCR -> Artifact-Registry (#878)

* cleanup: rm obsolete/redundant attributes

* switch to new testmachinery-image-location

* switch GCR -> Artifact-Registry

GCR has been deprecated [0] in favour of Artifact-Registry.

Thus, change push-targets for OCI-Images:

- europe-docker.pkg.dev/gardener-project/snapshots for snapshots
- europe-docker.pkg.dev/gardener-project/releases for releases
- europe-docker.pkg.dev/gardener-project/public for read-access to
  snapshots + releases

[0]
https://cloud.google.com/artifact-registry/docs/transition/transition-from-gcr

himanshu-kun committed:
reconcile on removal of critical component taint

himanshu-kun committed:
reconcile machine on specific node taint updates

elankath committed:
Proposal: Post Create Initialization (#868)

* WIP: draft for machine update

* changed design to post-creation logic

* fixed typo in background

* addressed review comments

* Driver.InitializeMachine added

* instance initialization flow

* re-orged section

sssash18 committed:
Fixed MCM restart during cleanup (#879)

* Fixed MCM restart during cleanup

* quick fix

* Addressed review comments

himanshu-kun committed:
adapted unit tests for create errors

sssash18 committed:
Made the gen-crd-api-reference-docs binary platform independent

himanshu-kun committed:
fixed some failing tests

sssash18 committed:
Quick fix

lookuptable committed:
Support GKE version schema

himanshu-kun committed:
Apply suggestions from code review

Co-authored-by: Aaron Francis Fernandes <79958509+aaronfern@users.noreply.github.com>
Co-authored-by: Rishabh Patel <66425093+rishabh-11@users.noreply.github.com>

himanshu-kun committed:
addresed Rishab's suggestion part 1

himanshu-kun committed:
addressed Rishab's review comments part 2

himanshu-kun committed:
renamed updateRetryRequired to updateRetryPeriod

aaronfern committed:
Merge pull request #881 from sssash18/gen-crd-fix

Made the gen-crd-api-reference-docs binary platform independent

aaronfern committed:
Merge pull request #877 from himanshu-kun/remove-status-update-reconcile-triggers

Reduce noisy reconciles + enhance logs

sssash18 committed:
Merge pull request #882 from lookuptable/gke-versions

Support checking GKE version schema

sssash18 committed:
Env var  for IT tests (#884)

* Doc update for IT tests

* Added Target Resource group

* Refactored sentence

* Update pkg/test/integration/common/framework.go

Co-authored-by: Himanshu Sharma <79965161+himanshu-kun@users.noreply.github.com>

* Doc update

---------

Co-authored-by: Himanshu Sharma <79965161+himanshu-kun@users.noreply.github.com>

elankath committed:
Check, update node label on machine obj prior to drain,termination (#887)

* Check, update node label on machine obj prior to drain

* added logs for node deletion

* modified log statement for clarity

himanshu-kun committed:
Some updates to FAQ (#886)

* prioritize deletion of machines FAQ

* added state transition diagram

* addressed review comments by Tarun

Kostov6 committed:
Introduce the use of central manifest (#847)

gardener-robot-ci-1 committed:
Release v0.51.0

gardener-robot-ci-1 committed:
Prepare next Dev Cycle v0.52.0-dev

elankath committed:
nil fix in MC.updateNodeToMachine (#893)

* nil fix in MC.updateNodeToMachine

* fix error log in updateNodeToMachine

---------

Co-authored-by: Rishabh Patel <rishabh.patel@sap.com>

sssash18 committed:
Added `Architecture` in NodeTemplate (#894)

* Added arch

* Added arch to nodetemplate

* Addressed review comments

elankath committed:
security fix: removal of go-git (#896)

* security fix: removal of go-git

* env harmonization: TARGET_RESOURCE_GROUP->TARGET_CLUSTER_NAME

* Update docs/development/integration_tests.md

Co-authored-by: Rishabh Patel <66425093+rishabh-11@users.noreply.github.com>

* doc: TARGET_CLUSTER_NAME defn move to general setup section

---------

Co-authored-by: Rishabh Patel <66425093+rishabh-11@users.noreply.github.com>

gardener-robot-ci-1 committed:
Release v0.52.0

gardener-robot-ci-2 committed:
Prepare next Dev Cycle v0.53.0-dev

RaphaelVogel committed:
Make repo compliant (#900)

elankath committed:
VM Instance Initialization: Driver.InitializeMachine (#898)

* New provider Op: Driver.InitializeMachine

* fixed godoc

* fix: handle NotFound properly in fake

* addressed second round of review-changes for code

* updated error code docs for InitializeMachine

* add  error code row for GetMachineStatus

* remmoved Lastknownstate

* removed accident check-in

* fixed accidental tyo

* fixed desc for codes.Internal for InitMachine

rishabh-11 committed:
Remove `vendor` directory (#903)

* prepare for vendor folder removal

* Remove vendor directory

* update dep5 file, licenses

* fix code generation

* remove vendor from add_license_headers

* use controller-gen 0.14.0

elankath committed:
Diagnostics Logs for MCM (#906)

* log when machine changes to FAILED phase

* added logs for priority annot & machine deletion

* adjusted logging

* refined log levels for some existing logs

* refined log levels for some existing logs

rishabh-11 committed:
Upgrade k8s dependencies to `v0.29.3`, Fix yaml parsing issues (#907)

* upgrade k8s dependencies to v0.29.3

* make check changes

* fix yaml parsing in integration tests

* address review comments

sssash18 committed:
Enhanced local IT setup (#899)

* Enhanced local IT setup

* Removed MCC from non-gardener-setup script

* Removed MCC from non-gardener-setup script

* Update hack/non_gardener_local_setup.sh

Co-authored-by: Aaron Francis Fernandes <79958509+aaronfern@users.noreply.github.com>

* Improved the IT test docs.

* Improved the IT test docs.

---------

Co-authored-by: Aaron Francis Fernandes <79958509+aaronfern@users.noreply.github.com>

Commit:
Release v0.53.0

gardener-robot-ci-1 committed:
Prepare next Development Cycle v0.54.0-dev

zkdev committed:
chore: Update Concourse URL (#916)

maboehm committed:
Update semver package to v3 (#917)

sssash18 committed:
fixes docker image

sssash18 committed:
fixes docker image

sssash18 committed:
fixes docker image

sssash18 committed:
fixes docker image

sssash18 committed:
Update Makefile

aaronfern committed:
Merge pull request #922 from sssash18/fix_docker_image

Fix docker image

RAPSNX committed:
Remove orphaned CRD (#923)

sssash18 committed:
Improved context with timeout for DrainNode (#920)

* improved drain timeout calculation

* removed select for context cancellation

* passing drain context only in drainNode

* removed effective drain timeout modification

* create drainContext in RunDrain

* remove time.Sleep

rishabh-11 committed:
fix api doc (#927)

rishabh-11 committed:
Allow deletion to proceed in case of VM initialization error (#928)

* allow deletion to proceed in case of VM initialization error

* omit tool binaries

* set_makefile_env: addec CONTROL_NAMESPACE, LEADER_ELECT

---------

Co-authored-by: elankath <tarun.ramakrishna.elankath@sap.com>

ScheererJ committed:
Add namespace to dummy machine object so that driver can rely on the namespace being set (#932)

aaronfern committed:
Update golang to 1.22.5 (#929)

* Update golang to 1.22.5

* Added /hack/tools/bin to gitignore

* Removed commented code

* Simplify a nested if condition

* Renamed an import

oliver-goetz committed:
Introduce a feature to propagate the machine name to user data and as label to the node (#919)

* Replace magic string `<<MACHINE_NAME>>` with machine name when bootstrapping

* Add `machine-name` to node labels

* Address PR review feedback

* Address PR review feedback v2

* Address PR review feedback v3

* Address PR review feedback v4

* Add tests

thiyyakat committed:
Fix gardener_local_setup script to use correct control cluster namespace (#935)

* Fixed gardener_local_setup.sh script to use the correct CONTROL_CLUSTER_NAMESPACE

* Update hack/gardener_local_setup.sh

Co-authored-by: Rishabh Patel <66425093+rishabh-11@users.noreply.github.com>

---------

Co-authored-by: Rishabh Patel <66425093+rishabh-11@users.noreply.github.com>

ary1992 committed:
Upgrade k8s.io/* to v0.31 (#938)

* Upgrade k8s.io/* to v0.31.0

* Adapt fakeclient

Co-authored-by: Martin Weindel <martin.weindel@sap.com>

* Fix unit test

* Adapt code generator

* Run make generate

* Upgrade golang-ci lint version

Co-authored-by: Martin Weindel <martin.weindel@sap.com>

* Make linter happy

Co-authored-by: Martin Weindel <martin.weindel@sap.com>

---------

Co-authored-by: Martin Weindel <martin.weindel@sap.com>

thiyyakat committed:
Modify Machine Creation flow to make sure node label is updated before initialization of VM. Modify Deletion flow to call DeleteMachine even if VM is not found. (#940)

* Change order in which labels are updated and machine is initialized

* Change how node label is updated in getVMStatus. Proceed to drain if VM not found

* Remove error log when initializeMachine is unimplemented

* Change where initializeMachine is called in triggerCreationFlow

* Correct GetVMStatus flow from previous commit

* Remove unnecessary comments

* Initialize VM and update labels in the same reconciliation

* Fix make check errors

* Address review comments

* Address review comments part 2

* Add comment for future changes to triggerCreationFlow to clean up dirty code

* Correct err variable assignment

* Shift comment to the right place

Commit:
Release v0.54.0

gardener-robot-ci-1 committed:
Prepare next Development Cycle v0.55.0-dev

acumino committed:
fix something (#942)

timebertt committed:
Don't track reattachment of provider-unrelated PVs (#937)

* Don't track reattachment of provider-unrelated PVs

* Add unit tests for `getPodVolumeInfos`

* Review: simplify `addAll`, drop `toListOfAny`

Commit:
fix ctx declaration for vol detachment and reattachment

Commit:
remove support for v1beta1 PDBs

n-boshnakov committed:
Fixed broken links - 2024.10 (#946)

unmarshall committed:
Merge pull request #947 from rishabh-11/remove-shadow

Fix context declaration for vol detachment and reattachment

thiyyakat committed:
Reduce load on etcd/kube-apiserver on pod eviction (#949)

* Reduce etcd traffic by using SharedInformer to list pods for drain logic

* Address review comments

oliver-goetz committed:
Export magic userdata strings (#953)

thiyyakat committed:
Introduce `gosec` for Static Application Security Testing (SAST) (#954)

* Introduce make targets for sast and address security issues.

* Address review comments

thiyyakat committed:
Fix edge-case where scale-from-zero of MCD is mistaken for a rolling update (#956)

* Fix edge-case where scale-from-zero is mistaken for rolling update

* Address review comments

* Address review comments of PR #954

* Address review comments

* Fix unit test

gardener-robot-ci-1 committed:
Release v0.55.0

gardener-robot-ci-1 committed:
Prepare next Development Cycle v0.56.0-dev

thiyyakat committed:
Fix installation of `gosec` and `addlicense` (#957)

* Change values and names of variables used in tools.mk file to fix bug in gosec and addlicense installation

* run make generate

* Update go version to v1.23.3

---------

Co-authored-by: Rishabh Patel <rishabh.patel@sap.com>

thiyyakat committed:
Adapt pipeline_definitions to include SAST linting logs in OCM descriptor (#959)

* Add newline at the end of files

* Adapt pipeline_definitions to include SAST linting logs in OCM descriptor

aaronfern committed:
Change integration test to add gardener-node-agent secret name to machine objects and MCDs. (#963)

Co-authored-by: thiyyakat <meghana.thiyyakat@gmail.com>

elankath committed:
Fix Race conditions in Targeted Deletion of machines by CA (#965)

* read TriggerDeletionByMCM on MCD and set MachinePriority=1 annotation on machines

* fixed dup import issue

* improved logging+annotation clear

* addressed final review comments

* removed todo, added unit-test binary files to .gitignore

* changed to better var names

* reverted code review change - nil/empty check is necessary

gardener-robot-ci-1 committed:
Release v0.56.0

gardener-robot-ci-1 committed:
Prepare next Development Cycle v0.57.0-dev

aaronfern committed:
Add a nil check before trying to adding mc annotation (#966)

acumino committed:
[GEP 31] Introduce API changes for supporting InPlaceUpdate (#962)

* Introduce APIs for InPlaceUpdate

Co-authored-by: Shafeeque E S <shafeeque.e.s@sap.com>
Co-authored-by: Ashish Ranjan Yadav  <ashish.ranjan.yadav@sap.com>

* Adapt API changes

Co-authored-by: Shafeeque E S <shafeeque.e.s@sap.com>
Co-authored-by: Ashish Ranjan Yadav  <ashish.ranjan.yadav@sap.com>

* Add validation for InPlaceUpdate strategy

Co-authored-by: Shafeeque E S <shafeeque.e.s@sap.com>
Co-authored-by: Ashish Ranjan Yadav  <ashish.ranjan.yadav@sap.com>

* Nits

Co-authored-by: Shafeeque E S <shafeeque.e.s@sap.com>
Co-authored-by: Ashish Ranjan Yadav  <ashish.ranjan.yadav@sap.com>

* Introduce labels constants for InPlaceUpdate

Co-Authored-By: Shafeeque E S <shafeeque.e.s@sap.com>
Co-Authored-By: Ashish Ranjan Yadav <ashish.ranjan.yadav@sap.com>

* Address Review

* Add test

* Address Review

* Address Review

* Address Review

* Address Review

---------

Co-authored-by: Shafeeque E S <shafeeque.e.s@sap.com>
Co-authored-by: Ashish Ranjan Yadav <ashish.ranjan.yadav@sap.com>

acumino committed:
[GEP-31] Introduce label constants and `MachineConfiguration` options for in-place updates  (#970)

* Introduce labels constants for InPlaceUpdate

Co-Authored-By: Shafeeque E S <shafeeque.e.s@sap.com>
Co-Authored-By: Ashish Ranjan Yadav <ashish.ranjan.yadav@sap.com>

* Introduce `MachineInPlaceUpdateTimeout` field

Co-authored-by: Shafeeque E S <shafeeque.e.s@sap.com>
Co-authored-by: Ashish Ranjan Yadav  <ashish.ranjan.yadav@sap.com>

* Introduce `DisableHealthTimeout` field

Co-authored-by: Shafeeque E S <shafeeque.e.s@sap.com>
Co-authored-by: Ashish Ranjan Yadav  <ashish.ranjan.yadav@sap.com>

* Run `make generate`

Co-authored-by: Shafeeque E S <shafeeque.e.s@sap.com>
Co-authored-by: Ashish Ranjan Yadav  <ashish.ranjan.yadav@sap.com>

* Nits

* Address Review

* Address Review

---------

Co-authored-by: Shafeeque E S <shafeeque.e.s@sap.com>
Co-authored-by: Ashish Ranjan Yadav <ashish.ranjan.yadav@sap.com>

gardener-robot-ci-1 committed:
Release v0.57.0

gardener-robot-ci-1 committed:
Prepare next Development Cycle v0.58.0-dev

n-boshnakov committed:
Transfer Frontmatter - 2025.02 (#969)

takoverflow committed:
Ensure cleanup of IT resources and processes (#968)

* refactor: extract cleanup fns & allow for cleanup in beforesuite

* fix: kill processes run by IT before and after running the suite

Iterates through a list of known processes for MC and MCM started
by the test suite and ensures they are terminated by calling `Kill()`
on them to prevent un-necessary failures due to "socket being used"
by the leftover processes.

* cleanup: remove CONTAINER_IMAGE checks and related fn

* refactor: added event based machineset freeze check instead of logs

* framework: add changes to support virtual provider

* Address review comments

* nit: rename to a sensible name

* Added constants for events, address review comments

* bugfix: add supported argument on fetching process to kill

* chore: update ginkgo version and flags

Ref: https://onsi.github.io/ginkgo/MIGRATING_TO_V2#changed-command-line-flags
Ref: https://onsi.github.io/ginkgo/MIGRATING_TO_V2#improved-profiling-support

takoverflow committed:
Introduce machine termination queue to handle deletions (#964)

* re-use `machineCreateErrorHandler` in `CreationFlow`

Rather than explicitly setting the LastOperation value
and updating the status for multiple error codes, fall
back to using the already available ErrorHandler to
clean up the creation flow.

* feat: introducing machine termination queue

Isolated the machine deletion flow into a separate workerqueue
that only processes terminating machines, the remaining machine
reconciliation flow happens as is in the `machineQueue`. The
machine watch functions for `update` and `delete` events are
updated to account for the introduction of the new queue.

* bugfix: check for `DeletionTimestamp` on ADD events

In case the MCM process stops during deletion, upon restart
the same terminating machine would be added to the normal
machine queue (as part of ADD event) and since its deletion
timestamp is set, it is not processed any further hence the
deletion never proceeds. By checking for a terminating machine
on ADD events, this is resolved.

* nit: remove unnecessary guard around triggerDeletionFlow

* Address review comments

* machine.go: add tombstone handline for machine delete events
* machine.go: remove redundant check in creationflow
* machine_util.go: added docstring for createErrorHandler

* review: added check for successful casting

* nit: add explanatory comments and improve logged errors

* review: requeue on successful deletion as well

elankath committed:
Trigger removal of CA tainted machine without waiting for replica change (#972)

* for-deletion machines dont want for replica change

* fix timing issue in additional unit test

* added unit test for managereplicas

* fixed ineffectual assignment&unused

* added additional check in test

* fixed test assertion

* addressed review comments

* accepted review comment

---------

Co-authored-by: elankath <tarun@elankath.com>

takoverflow committed:
Update golangci-lint version (#978)

ary1992 committed:
[GEP-31] Introduce Constants needed for Conditons and Status for InPlace update  (#980)

* Introduce Constants needed for Conditons and Status

Co-Authored-By: Shafeeque E S <shafeeque.e.s@sap.com>
Co-Authored-By: Ashish Ranjan Yadav <ashish.ranjan.yadav@sap.com>

* Address PR review feedback

---------

Co-authored-by: Sonu Kumar Singh <sksgkpvks@gmail.com>
Co-authored-by: Shafeeque E S <shafeeque.e.s@sap.com>

acumino committed:
[GEP-31] Controller changes to Support `Auto` `InPlaceUpdate`  strategy (#973)

* Replicas calculation for new machine set in case of InPlace

Co-Authored-By: Shafeeque E S <shafeeque.e.s@sap.com>
Co-Authored-By: Ashish Ranjan Yadav <ashish.ranjan.yadav@sap.com>

* Add utility function

Co-Authored-By: Shafeeque E S <shafeeque.e.s@sap.com>
Co-Authored-By: Ashish Ranjan Yadav <ashish.ranjan.yadav@sap.com>

* [MCD] Use AutoInPlace rolling strategy

Co-Authored-By: Shafeeque E S <shafeeque.e.s@sap.com>
Co-Authored-By: Ashish Ranjan Yadav <ashish.ranjan.yadav@sap.com>

* [MCD] Label all old nodes and machine as candidate for update

Co-Authored-By: Shafeeque E S <shafeeque.e.s@sap.com>
Co-Authored-By: Ashish Ranjan Yadav <ashish.ranjan.yadav@sap.com>

* [MCD] Add scale up logic

Co-Authored-By: Shafeeque E S <shafeeque.e.s@sap.com>
Co-Authored-By: Ashish Ranjan Yadav <ashish.ranjan.yadav@sap.com>

* [MCD] Add scale down logic

Co-Authored-By: Shafeeque E S <shafeeque.e.s@sap.com>
Co-Authored-By: Ashish Ranjan Yadav <ashish.ranjan.yadav@sap.com>

* [MCD] Sync MachineSets after updates and on new reconciliation

Co-Authored-By: Shafeeque E S <shafeeque.e.s@sap.com>
Co-Authored-By: Ashish Ranjan Yadav <ashish.ranjan.yadav@sap.com>

* [MCD] Handle max surge case

Co-authored-by: Shafeeque E S <shafeeque.e.s@sap.com>
Co-authored-by: Ashish Ranjan Yadav  <ashish.ranjan.yadav@sap.com>

* [MCD] Label all the old machine sets to disable the scale up

Co-Authored-By: Shafeeque E S <shafeeque.e.s@sap.com>
Co-Authored-By: Ashish Ranjan Yadav <ashish.ranjan.yadav@sap.com>

* Enhance testing lib for more use cases

Co-Authored-By: Shafeeque E S <shafeeque.e.s@sap.com>
Co-Authored-By: Ashish Ranjan Yadav <ashish.ranjan.yadav@sap.com>

* Add test for in-place updates

Co-Authored-By: Shafeeque E S <shafeeque.e.s@sap.com>
Co-Authored-By: Ashish Ranjan Yadav <ashish.ranjan.yadav@sap.com>

* [MCS] Ignore in-place updated machine in new machine set

Co-authored-by: Shafeeque E S <shafeeque.e.s@sap.com>
Co-authored-by: Ashish Ranjan Yadav  <ashish.ranjan.yadav@sap.com>

* [MCS] Label all oldISs to not upscale for the transferred machine

When transferring ownership old machine set should not create new machine for the machine transferred to new machine set

Co-authored-by: Shafeeque E S <shafeeque.e.s@sap.com>
Co-authored-by: Ashish Ranjan Yadav  <ashish.ranjan.yadav@sap.com>

* [Machine Safety] Calculate maxSurge based on update strategy

Co-authored-by: Shafeeque E S <shafeeque.e.s@sap.com>
Co-authored-by: Ashish Ranjan Yadav  <ashish.ranjan.yadav@sap.com>

* [MC] Update node condition based on the in-place update state

Co-Authored-By: Shafeeque E S <shafeeque.e.s@sap.com>
Co-Authored-By: Ashish Ranjan Yadav <ashish.ranjan.yadav@sap.com>

* [MC] Drain Nodes which has been selected for update

Co-Authored-By: Shafeeque E S <shafeeque.e.s@sap.com>
Co-Authored-By: Ashish Ranjan Yadav <ashish.ranjan.yadav@sap.com>

* [MC] Adapt machine health reconciliation

Co-Authored-By: Shafeeque E S <shafeeque.e.s@sap.com>
Co-Authored-By: Ashish Ranjan Yadav <ashish.ranjan.yadav@sap.com>

* [MC] Trigger  machine reconciliation on addition of in-place update node label

Co-authored-by: Shafeeque E S <shafeeque.e.s@sap.com>
Co-authored-by: Ashish Ranjan Yadav  <ashish.ranjan.yadav@sap.com>

* Nit

* Fix case where there are zero machines in oldMachineSets

* Address Review

* Address Review

* Address Review

* Address PR review feedback

* Adapt constant name change

* Address Review

* Address Review

* Continue in case of node label is not present on the machine

---------

Co-authored-by: Shafeeque E S <shafeeque.e.s@sap.com>
Co-authored-by: Ashish Ranjan Yadav <ashish.ranjan.yadav@sap.com>

ary1992 committed:
[GEP-31] Controller changes to Support Manual InPlaceUpdate strategy (#987)

* Manual Inplace

Co-Authored-By: Shafeeque E S <shafeeque.e.s@sap.com>
Co-Authored-By: Sonu Kumar Singh <28653770+acumino@users.noreply.github.com>

* Minor Nit

Co-Authored-By: Shafeeque E S <shafeeque.e.s@sap.com>
Co-Authored-By: Sonu Kumar Singh <28653770+acumino@users.noreply.github.com>

* Address Review

* Address Review

* Get MCD for machines without passing context

* Add predicate to reconcile MCD when machine update is successful

* Enhance few logs

* Change is to machineSet

---------

Co-authored-by: Shafeeque E S <shafeeque.e.s@sap.com>
Co-authored-by: Sonu Kumar Singh <28653770+acumino@users.noreply.github.com>
Co-authored-by: Sonu Kumar Singh <sksgkpvks@gmail.com>

takoverflow committed:
Minor fixes (#981)

* Update docstring for `MachineDeploymentFrozen`

* Change retryPeriod for machines failing creation due to ResourceExhaustion

renormalize committed:
Fix version logging at binary startup. (#985)

* Fix version logging at binary startup.

Signed-off-by: Saketh Kalaga <51327242+renormalize@users.noreply.github.com>

* Address review comments from @aaronfern.

* Add `ldflags` to the `start` make target.

---------

Signed-off-by: Saketh Kalaga <51327242+renormalize@users.noreply.github.com>

gardener-robot-ci-1 committed:
Release v0.58.0

gardener-robot-ci-1 committed:
Prepare next Development Cycle v0.59.0-dev

aaronfern committed:
Upgrade base image to debian12 (#991)

elankath committed:
added VirtualCapacity to NodeTemplate

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/gardener/machine-controller-manager?pullRequest=998) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/gardener/machine-controller-manager?pullRequest=998) before we can accept your contribution.<br/>**12** out of **13** committers have signed the CLA.<br/><br/>:white_check_mark: n-boshnakov<br/>:white_check_mark: thiyyakat<br/>:white_check_mark: unmarshall<br/>:white_check_mark: acumino<br/>:white_check_mark: takoverflow<br/>:white_check_mark: elankath<br/>:white_check_mark: aaronfern<br/>:white_check_mark: oliver-goetz<br/>:white_check_mark: renormalize<br/>:white_check_mark: timebertt<br/>:white_check_mark: Kumm-Kai<br/>:white_check_mark: ary1992<br/>:x: Rishabh Patel<br/><hr/>**Rishabh Patel** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/gardener/machine-controller-manager?pullRequest=998) it.</sub>

aaronfern committed:
fetch pod from cache before eviction is attempted (#979)

takoverflow committed:
Don't update machine status to terminating on deletion failure (#995)

aaronfern reviewed (commented):
You would also need to add the field to `types.go` here: https://github.com/gardener/machine-controller-manager/blob/master/pkg/apis/machine/types.go#L737

Kumm-Kai committed:
fix: use `strconv.ParseBool` for `force-deletion` label (#999)

aaronfern committed:
Add node_name label to the mcm_machine_info metric (#1000)

timebertt committed:
Simplify local development (#1003)

* Simplify local development

Push an image to the local setup registry using:
```
make docker-image IMAGE_REPOSITORY=garden.local.gardener.cloud:5001/machine-controller-manager PUSH=true
```

* Speed up repetetive image builds

elankath committed:
Merge branch 'master' into virtcap

takoverflow committed:
Add validation for flags passed to machine controller (#1006)

aaronfern committed:
Add validations for data passed via flags (#1005)

timebertt committed:
[GEP-28] Support managing machines without target cluster for `gardenadm bootstrap` (#1004)

* Add docs for running without target cluster

* Handle disabled target cluster in manager/controller/test setup

* Adapt `MachineDeployment` controller

We skip any interaction with the target cluster by returning an error early if `InPlaceUpdate` is configured when running without a target cluster.
`InPlaceUpdate` only works with a target cluster – `Node` metadata needs to be updated for it.
Apart from that, there is no interaction with the target cluster in the `MachineDeployment` controller.

* Adapt safety controller

If running without a target cluster, we can entirely disable the SafetyAPIServer and Node controllers.
Also, there is no need to annotate `Nodes` unmanaged by machine-controller-manager.

* Adapt machine creation flow

During machine creation, we never set the `node` label on the `Machine` to disable all interaction with the target cluster.
With this, we skip waiting for the `Node` object to be registered.
We also don't create any bootstrap tokens.
Lastly, instead of tranisitioning to `Pending`, the `Machine` transitions to `Available` after creating the VM.

* Adapt machine deletion flow

If running without a target cluster, we skip drain and volume attachment logic and go straight to VM deletion.
The `node` label does not exist, so we won't delete the `Node` object.

* Extract `constants.TargetKubeconfigDisabledValue`

* Move flag validation

elankath committed:
Merge branch 'master' into virtcap

elankath committed:
regenerated apis

elankath committed:
regenerated apis

elankath committed:
added sync for virtual capacity

elankath committed:
added sync for virtual capacity

aaronfern reviewed (commented):
Thanks for the PR @elankath!
A few comments from me

elankath committed:
Update pkg/util/provider/machinecontroller/machine_util.go

Co-authored-by: Aaron Francis Fernandes <79958509+aaronfern@users.noreply.github.com>

elankath committed:
initializedNodeAnnotation check added
